lib/AuthenticationSDK/util/MLEUtility.rb

```diff
@@ -0,0 +1,115 @@
+require_relative '../logging/log_factory.rb'
+require 'jose'
+public
+  class MLEUtility
+    @log_obj
+    def self.check_is_mle_for_API(merchant_config, is_mle_supported_by_cybs_for_api, operation_ids)
+      is_mle_for_api = false
+      if is_mle_supported_by_cybs_for_api && merchant_config.useMLEGlobally
+        is_mle_for_api = true
+      end
+      if merchant_config.mapToControlMLEonAPI.nil? && merchant_config.mapToControlMLEonAPI.nil?
+        operation_ids.each do |operation_id|
+          if merchant_config.mapToControlMLEonAPI.key?(operation_id)
+            is_mle_for_api = merchant_config.mapToControlMLEonAPI[operation_id]
+            break
+          end
+        end
+      end
+      is_mle_for_api
+    end
+
+    def encrypt_request_payload(merchant_config, request_payload)
+      if request_payload.nil?
+        return nil
+      end
+      @log_obj = Log.new(merchant_config.log_config, 'MLEUtility')
+      @log_obj.logger.info('Encrypting request payload')
+      @log_obj.logger.debug('LOG_REQUEST_BEFORE_MLE: ' + request_payload)
+
+
+      begin
+        certificate = get_certificate(merchant_config, @log_obj)
+        validate_certificate(certificate, merchant_config, @log_obj)
+        serial_number = extract_serial_number_from_certificate(certificate)
+        if serial_number.nil?
```

Comment:
if serial number is null then we can put default serial number from cert as same as implemented in Java.